### ISSUE TITLE
分布式 PIPELINE: 缺 /distributed/decode 端点 — forward 链无 lm_head/出 token

### ISSUE THREAD

**Issue by dahai80** [dahai80/fusion-mlx#630]
## 问题

`fusion_mlx/api/distributed_routes.py` 现有 5 端点 (load_shard / pipeline_step / sync_weights / shards / drop_shard) 只覆盖 **前向层链**, 不覆盖 **lm_head + decode**。导致 fusion-multi-node Pipeline Parallelism 能跑通层切分链 (层 [start,end) 前向, b64.npy 激活跨节点), 但 **末节点出口只拿到 hidden_states, 永不出 token**。

## 现状 (shard.py:234-288)

`pipeline_step` 末尾:
```python
for i in range(start, end):
    hidden = layers[i](hidden)
mx.eval(hidden)
return {"hidden_states": ..., "shape": ..., "dtype": ...}
```

到 `layers[end]` 为止, **从未应用 `inner.lm_head`** (或 tied embedding output projection)。hidden → vocab logits → argmax/sampling → next token 这步不存在。

## 影响

fusion-multi-node PIPELINE 模式 = 硬阻断: 集群能调度 + 切层 + 跨节点传激活, 但产不出推理结果。lm_head 权重在 fusion-mlx 侧 (model 已加载), 调度器 (fusion-multi-node) 无法本地加载模型权重做 decode — 必须由 fusion-mlx 暴露端点。

## 建议方案

新增端点:

```
POST /distributed/decode
  req: { shard_id, hidden_states: b64.npy, temperature?: float, top_p?: float, max_tokens?: int }
  resp: { token_ids: list[int], logits?: b64.npy (可选), shape, dtype }
```

实现要点:
- 末 shard (layer_range 末段) 调用; 收 hidden_states → `inner.lm_head(hidden)` → logits → argmax (greedy) 或 sampling
- tied embedding 模型: lm_head 复用 `inner.embed_tokens.weight.T`
- 与 pipeline_step 同 shard 注册机制 (load_shard 已加载 model, lm_head 权重在内存)
- 复用现有 b64.npy 序列化 (`serialize_activation`/`deserialize_activation`), logits 大可不传 (仅返 token_ids 省带宽)

## 协调

- fusion-multi-node H1 硬阻断, 等此端点落地后接 `FusionMLXBackend.decode` + `ClusterMaster._dispatch_pipeline` 末节点出口改调 decode
- 现网单机 fusion-mlx 推理不受影响 (走正常 /v1/chat/completions), 仅分布式 PIPELINE 模式缺

**Comment by dahai80** [comment]
Closed by #638 (merged 58c6f449). `pipeline_step` ran only the layer loop and returned the un-normed post-layer hidden states — the final `inner.norm` + `lm_head` step that produces vocab logits was never exposed, so distributed PIPELINE mode could slice layers across nodes but never produced a token. `POST /distributed/decode` now applies `inner.norm` then the output projection (tied embeddings → `embed_tokens.as_linear`; else `model.lm_head`) and samples via `mlx_lm make_sampler` (temp 0/None = greedy argmax). Verified on `Llama-3.2-1B-Instruct-4bit`: decode reproduces the un-split full forward's argmax bit-exactly (5 real-model tests). docs/distributed-pipeline.md + README updated. Note: multi-token autoregressive decode (the issue's `max_tokens`) is deliberately out of scope — `decode` is the single-forward building block; the scheduler composes `pipeline_step`+`decode` across nodes for multi-token output, and KV-cache threading is a future streaming-scheduler concern.
